lib/queue.rb

```diff
@@ -1,16 +1,42 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(10)
+    @front = -1 
+    @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    #Circular buffer method
+    #Check if queue is empty
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1 
```

Comment:
Since you're going to advance @back one step later in this method better to set it to 0